runtime-common/search-index.ts

```diff
@@ -616,6 +617,46 @@ export class SearchIndex {
     );
   }
 
+  private buildSorter(
+    expressions: Sort | undefined
+  ): (e1: SearchEntry, e2: SearchEntry) => number {
+    if (!expressions || expressions.length === 0) {
+      return () => 0;
+    }
+    let sorters = expressions.map(({ by, direction }) => {
+      return (e1: SearchEntry, e2: SearchEntry) => {
+        let a = e1.searchData[by];
+        let b = e2.searchData[by];
+        if (a === undefined) {
+          return direction === "desc" ? -1 : 1; // if descending, null position is before the rest
+        }
+        if (b === undefined) {
+          return direction === "desc" ? 1 : -1; // `a` is not null
+        }
+        if (typeof a === "string") {
+          a = a.toLowerCase();
+          b = b.toLowerCase();
+        }
```

Comment:
This block of code can stay in this PR, but once this is landed and https://github.com/cardstack/runtime-spike/pull/92 is landed, it gets replaced with `getQueryableValue` so that different field cards can do different things here. A string being case-insensitive is only one example of the possible sorts people would want.

(I think getQueryableValue is sufficient and we don't need a separate getSortableValue.)